### ISSUE TITLE
fighters will dock in flagship even when sorted to dock in other carriers

### ISSUE THREAD

**Issue by Vonpiper** [endless-sky/endless-sky#7502]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Originally posted by wuzinhis below
https://steamcommunity.com/app/404410/discussions/0/3473990693225567845/

Fighters will dock and undock from the player's flagship.  This happens when the ships are sorted on the "player info" screen such that fighters are directly below another carrier, not the player's flagship.

### Steps to Reproduce

Fly a flagship with fighter bays, such as the Kestrel or Bacitrain.
Have another ship in fleet with fighter bays.
Have some fighters in fleet like petrels.

While docked, open player info screen.
Sort flagship first, then the other carrier ship, then a fighter or two.
Ensure all ships are not parked.
Undock and press the deploy/recall fighters button.

Notice the fighters deploy from player's flagship.
Press deploy/recall again.
Notice fighters recall to the player's flagship.

### Expected Behavior

Instead of the fighters deploying and recalling to the player's flagship, they should deploy and recall to the carrier ship that they are directly below in the player info screen.

### Screenshots

Images below are from wuzinhis in this thread on steam:
https://steamcommunity.com/app/404410/discussions/0/3473990693225567845/

![image](https://user-images.githubusercontent.com/97364305/199403112-03b12fe0-72c2-43f6-bd75-31f96c60470d.png)

![image](https://user-images.githubusercontent.com/97364305/199403129-1722562e-5e67-406e-9e0e-b25a838a8206.png)

### Operating System

Windows 10 Pro v10.0.19044 build 19044

### Game Source

Steam

### Game Version

0.9.16.1

### Additional Information

I am able to reproduce the behavior that wuzinhis describes in the below thread on steam, but I am also using some mods.

https://steamcommunity.com/app/404410/discussions/0/3473990693225567845/

**Comment by Vonpiper** [comment]
I would like to try contributing a fix, but I'm new to large projects like this and I'm not sure where to start looking first.  I assume there is a getting-started page for contributors and bug hunters somewhere?

**Comment by 1010todd** [comment]
There is this page which devs generally recommend people to look at;

https://github.com/endless-sky/endless-sky/blob/master/readme-developer.md

There's also some code guideline or something in the repository wiki.

**Comment by vitalchip** [comment]
This is intentional behaviour. Ordering the ships in the Playe Info tab isn't meant to control where carried ships are loaded. It does have an effect but it's more coincidental than intended.

Two lists are created, one of carriers and one of carried ships, then the list of carried ships is sorted to move any which have jump capability to the bottom as their need for a bay is less than ships with no jump capability.

So any ordering on the Player Info can be undone by this step alone.

Then each carried ship is tested in turn for a suitable bay for each carried ship (from the sorted list), naturally the flagship is first.

To achieve what you're looking for we would probaly need some sort of loading interface where you could drag carried ships over a carrier, which would then check for available bays (type and quantity) and parent the ships at that time.

A simpler but less comprehensive option would be a toggle to skip the flagship until all other carriers were loaded.

**Comment by ThrawnCA** [comment]
Is there any scenario where it matters which ship a fighter docks in, apart from knowing which fighters you'll lose if a ship is destroyed? Which sounds like a level of micro-managing that ES doesn't normally bother with.

**Comment by Vonpiper** [comment]
> Is there any scenario where it matters which ship a fighter docks in, apart from knowing which fighters you'll lose if a ship is destroyed? Which sounds like a level of micro-managing that ES doesn't normally bother with.

I'm interested in the penalty to agility when fighters are docked.

**Comment by vitalchip** [comment]
I can imagine scenarios where it could be advantageous to have better control over loading.

If you had a few carriers but didn't have a full complement of fighters then it would be better for the fighters to be distributed rather than fully loading each carrier in turn. This caan help keep ship agility somewhat similar between carriers and means a ship which ends up out of formation might have at least a couple of fighters for defence rather than them all being in the flagship which is too far away to help.

It probably isn't terribly important as things stand but if the game ever moves in a direction where strategy matters then it could be a good supporting mechanism.

**Comment by petervdmeer** [comment]
The UI is indeed a bit confusing, but let's handle this as a missing feature, because nobody ever created any code yet to perform the fine-grained fighter loading control that vitalchip suggested in the message above this one.

**Comment by ThrawnCA** [comment]
If the ability to micromanage fighters is introduced, it would be important to ensure that micromanaging isn't compulsory. Eg assign fighters to the next available carrier by default when they're purchased/captured, and/or provide a button to default-assign every undocked fighter.

**Comment by Vonpiper** [comment]
Yeah this was meant to be super simple. I'd rather an escort be chunky than my flagship. Nothing more than that. Thank you for changing the labels and stuff.
